### PR TITLE
Use single PVC per namespace for pipelineRun build service components

### PR DESCRIPTION
**Pull request by AndrienkoAleksandr**
Use single PVC per namespace for pipelineRun build service components

Small demo:

https://www.youtube.com/watch?v=zAPaqh4FH-M

Signed-off-by: Oleksandr Andriienko <oandriie@redhat.com>

### CODE REVIEW

File: components/gitops/.tekton/trigger-template.yaml [hunk 1]
```diff
@@ -31,12 +31,6 @@ spec:
...
-                accessModes:
-                  - ReadWriteOnce
-                resources:
-                  requests:
-                    storage: 10Mi
-                #storageClassName: gp2
-                volumeMode: Filesystem
+            persistentVolumeClaim:
+              claimName: app-studio-default-workspace
+            subPath: gitops-service-$(tt.params.git-revision)
```
Comment (sbose78) [inline components/gitops/.tekton/trigger-template.yaml:36]: Can we do a `gitops-service/$(tt.params.git-revision)` ?

File: components/has/.tekton/trigger-template.yaml [hunk 2]
```diff
@@ -31,12 +31,6 @@ spec:
...
-                  - ReadWriteOnce
-                resources:
-                  requests:
-                    storage: 10Mi
-                #storageClassName: gp2
-                volumeMode: Filesystem
\ No newline at end of file
+            persistentVolumeClaim:
+              claimName: app-studio-default-workspace
+            subPath: application-service-$(tt.params.git-revision)
```
Comment (sbose78) [inline components/has/.tekton/trigger-template.yaml:36]: Can we do `application-service/$(tt.params.git-revision)`

Comment (sbose78) [reply components/has/.tekton/trigger-template.yaml:36]: though we can leave it as is since this workspace only has a single component anyway.

**Comment by sbose78** [comment]
@jduimovich is it fine if we merge this and then refactor the sub path in a follow-up PR ? I'd like to clean up the PVs lying around today manually and then have this go in. :) 

> The build-tag is a timestamp

Curious, how would the timestamp's placeholder be passed onto the resource template's workspace subpath?

**Comment by jduimovich** [comment]
Yes, this is fine to merge. The cleanup task can be updated in a follow-on PR.

Not sure we can generate the timestamp in a trigger, but the scripts I did make a pretty name, as I find the generateName: to be awful

**Review by jduimovich — APPROVED** [review]
LGTM
- follow on update todo for cleanup task to compute the subPath in a GC

File: components/has/.tekton/pvc.yaml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: app-studio-default-workspace
+spec:
+  accessModes:
+  - ReadWriteOnce
+  resources:
+    requests:
+      storage: 1Gi
```
Comment (goldmann) [inline components/has/.tekton/pvc.yaml:10]: How do we want control the size of the workspace? The default 1GB can be sufficient for many projects, but some will require much more space, especially if there are a few runs left on the workspace.

Comment (sbose78) [reply components/has/.tekton/pvc.yaml:10]: Good point @goldmann .
We could come up with a guidance ( which we'll need anyway for our users ) - but at the moment, I would leave it to the component owners to make the right call.

**Review by sbose78 — APPROVED** [review]

**Pull request merged**